include/boost/math/differentiation/autodiff.hpp

```diff
@@ -1317,6 +1347,14 @@ fvar<RealType,Order> atan(const fvar<RealType,Order>& cr)
     }
 }
 
+template<typename RealType1, size_t Order1, typename RealType2, size_t Order2>
+promote<fvar<RealType1,Order1>, fvar<RealType2,Order2>> atan2(const fvar<RealType1,Order1>& cr1, const fvar<RealType2,Order2>& cr2)
+{
+  using std::atan2;
+  return atan2(static_cast<RealType1>(cr2), static_cast<RealType2>(cr1));
```

Comment:
1. Should cr2 and cr1 be swapped?
2. Either way, how does this calculate derivatives? (Any overloaded mathematical function that returns an `fvar<>` type in this namespace is expected to calculate arbitrarily high derivatives accurately.)

Reply:
I needed to add type casting specializations after I added atan2 (which is called by one of the special functions). I’ll check if the type casting specializations remove the need for having this function at all.

Reply:
Ok. I just wanted to be sure I wasn't missing something. I was intending at some point to add this particular overload anyway, so no need to fix it, but a comment that it is known to be incorrect, or a TODO: calculate derivatives accurately, might be useful.

More generally, the primary motivation is/was to investigate the current compatibility between autodiff and boost's special_functions, and what is needed to acheive compatibility. Is it possible to give an overall summary? Or are there too many details to where one should just simply look through the tests you've made?

Btw I like the methods you added to autodiff.hpp, thank you.

Reply:
Unfortunately, relying the type conversion specializations won't work; the owens_t_hpp test breaks. Below is an explanation for having that function available in autodiff.hpp (based on `owens_t`'s reliance on it). I'll summarize 
`atan2` is called from `boost::math::owens_t` (specifically `owens_t_T6` from `owens_t_dispatch` as follows:
            `const RealType r = atan2(y, static_cast<RealType>(1 + a) );`
where `y` is the result of `owens_t_znorm2( h )` and `a`, with the original call being `boost::math::owens_t(h, a)`, with `h` and `a` being of type `RealType`.

With regard to `cr2` and `cr1` being swapped within the function, I relied the way `std::atan2` accepts arguments (`y` before `a`). I'll add a test for this to make sure the order in which I'm accepting arguments is consistent with `std::atan2` (and `boost::math::owens_t`).

I'll add a comment to the function regardless.